WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationCommentDetailViewController.swift

```diff
@@ -166,11 +166,11 @@ private extension NotificationCommentDetailViewController {
 
         // Refresh the current content if the underlying view controller supports it
         // Else, remove the existing child view controller and add a new one.
-        if let commentDetailViewController {
+        if let commentDetailViewController, comment.allowsModeration() {
```

Comment:
Instead of just relying on `comment.allowsModeration()` to decide if we should reload the view or not, using the content feels more deterministic especially if we add more conditions in the future.
```suggestion
        let newContent = makeNewContent(with: comment, notification: notification)
        if let commentDetailViewController, case .commentDetails = newContent {
            commentDetailViewController.refreshView(comment: comment, notification: notification)
        } else {
            self.content?.viewController.remove()
            let viewController = newContent.viewController
            ...
        }
```